### DISCUSSION TITLE
Why are the Honeywell 6160 keypads LCD backlight on since the install of the integration?

### DISCUSSION THREAD

**Discussion by AlexandreUSA** [Dilbert66/esphome-vistaECP#28] [Q&A] [answered]
After install of the "vistaalarm" integration and the electronics board, everything since to be working fine expect the LCD backlight of my Honeywell 6160 keypads (including one RF). The keypad LCD backlight is staying on all the time (on my 3 keypads). When I turn off the "vistaalarm connection", the behavior of the keypad backlight is back to normal (turns off after 30 to 40 seconds).

**Comment by Dilbert66** [comment, marked as answer]
Comment out line 460 (the "vista.write") line by putting to forward slashes in front as shown below.    The reason the lcd is lit is that my code sends an "*" every 30 seconds to force the display to show faults.  You can turn that off.  Try it and see if that takes care of th e issue.   It should not have too much impact in operation with it commented out.

Line 460
```
     if (!firstRun && vista.keybusConnected && millis() - asteriskTime > 30000  && !vista.statusFlags.armedAway && !vista.statusFlags.armedStay && !vista.statusFlags.programMode) {
            asteriskTime=millis();
        //    vista.write('*'); //send a * cmd every 30 seconds to cause panel to send fault status  when not armed
         
    }
```

**Reply by AlexandreUSA** [reply]
Thanks. I did comment out the line and the backlight is now turning off. This is good. 

However, there is something not right happening. When I clear the faults, I am back to Ready to Arm display. When I then trigger a zone, the "Hit * for faults" shows up as expected. However, without doing anything else, the fault will then show up on the screen with the backlight on for 30 seconds. After that the backlight turns off and continues to show the various faults without turning back on (in a way, okay). This is a different behavior than without the integration where the "Hit *" just disappear once the fault goes away.

So now, I do not have the "Hit * for faults" anymore (only for the first instance) and the backlight is turning off despite showing the faults. I kind of like this behavior as you can see the faults right away and you do not get the constant backlight on (while sleeping or during the day). When backlight is needed, just press the *. While I like it, I would like to understand the reason as this is a different behavior than my system without the vistaalarm integration.

**Comment by Dilbert66** [comment]
Unfortunately with the Vista20 there is no way for the emulation to detect a zone open status until the FAULT display is sent to the keypad.  It is a very basic system this way.   What it means then is that the ESP code and home assistant will not be able to know that a fault is active .. ie A door open if you have "Hit "*" for  faults showing on the keypad until a person goes and actually presses the * key.   For this reason to be able to detect a fault quickly and display it, my code sends an automatic * in response to the Hit * display to show the faults and be able to act on them.  You can turn that off and see if that works better for you by commenting the two lines below (around line 883 in vistaalarm.h)

```
     //       if (strstr(vista.statusFlags.prompt,"Hit *")) 
   //            vista.write('*');
```

**Reply by AlexandreUSA** [reply]
Interesting concept. Thanks for the explanation. Now, I am wondering why in the first case the vista.write('*'); turns on the backlight and not in the second case.

**Reply by AlexandreUSA** [reply]
I think I got it. The second vista.write('*') is only sent once when the Hit * appears. So the backlight only comes once. Make sense to me after looking at the code. Just tested it here and it does work as intended now. I just prefer your method both on the keypad and for HA integration. Maybe this could be an option in the yaml file to remove either case 1: turn off backlight or case 1 and 2 (no backlight and the need to press * on the keypad to start the pulling). 

So now, why having the first case? Sending the * command every 30 seconds if you send the * command when the Hit * appears anyway? What am I missing by commenting the first case only?

**Comment by Dilbert66** [comment]
You need to have the esp send the * when it sees the Hit * for faults or it would be pointless to have it there since it would never see the faults.  No option required.
